src/cascadia/QueryExtension/ILMProvider.idl

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+namespace Microsoft.Terminal.Query.Extension
+{
+    interface ILMProvider
+    {
+        // chat related functions
+        void ClearMessageHistory();
+        void SetSystemPrompt(String systemPrompt);
+        void SetContext(IContext context);
+
+        Windows.Foundation.IAsyncOperation<IResponse> GetResponseAsync(String userPrompt);
+
+        // auth related functions
+        void SetAuthentication(Windows.Foundation.Collections.ValueSet authValues);
+        event Windows.Foundation.TypedEventHandler<ILMProvider, String> AuthChanged;
+    }
+
+    interface IResponse
+    {
+        String Message { get; };
+        Boolean IsError { get; };
```

Comment:
Are there different kinds of errors? If so, how will the user of this API differentiate them?

I expect that if no auth is set up, we want to direct users to the auth flow. If the server responds with a "no capacity" error, we don't.

Reply:
> I expect that if no auth is set up, we want to direct users to the auth flow. If the server responds with a "no capacity" error, we don't.

The "message" parameter in the object is what directs the user to the next step (say to set up auth for example or to tell the user that a server error occurred)

Reply:
I'll be honest - I don't think that counts. There is a big difference between the string "Your authentication has expired, please go into the settings yourself and fix it" and a button that they can click to launch into setings.

The problem with a string-based API is that it is _impossible_ in the general case to present the user with anything more than inert text.

Reply:
Makes sense, added!